File: 文件夹/C#/排序/快速排序.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class 快速排序 : MonoBehaviour
{
    void QuickSort(int[] dataArray, int left, int right)//left开始索引 right结束索引
    {
        if (left < right)
        {
            int x = dataArray[left];//基准数
            int i = left;
            int j = right;

            while (true && i < j)
            {
                //从前往后，找到一个比基准数小的数字
                while (true && i < j)
                {
                    if (dataArray[j] <= x)//找到一个比基准数小数字
                    {
                        dataArray[i] = dataArray[j];
                        break; ;
                    }
                    else
                    {
                        j--;
                    }
                }
                //从后往前，找到一个比基准数大的数字
                while (true && i < j)
                {
                    if (dataArray[i] > x)
                    {
                        dataArray[j] = dataArray[i];
                        break; ;
                    }
                    else
                    {
                        i++;
                    }
                }
            }
            dataArray[i] = x;//left - i - right
            QuickSort(dataArray, left, i - 1);
            QuickSort(dataArray, i + 1, right);
        }
    }
}

File: 文件夹/Lua/排序/快速排序.lua
function QuickSort(array,left,right)
	if(left<right) then
		local x = array[left]
		local i = left
		local j = right
		while(i<j) do
			while(i<j) do
				if(array[j] <x) then
					array[i] = array[j]
					break
				else
					j = j-1
				end
			end
			while(i<j) do
				if(array[i] > x) then
					array[j] = array[i]
					break
				else
					i = i+1
				end
			end
		end
		array[i] = x;
		QuickSort(array,left,i-1)
		QuickSort(array,i+1,right)
	end
end
--array1ÅÅĞò
local array1 = {73,23,18,92}
QuickSort(array1, 1,#array1)
--array2ÅÅĞò
array2 = {73,23,18,92}
table.sort(array2)
--array1ÅÅĞòÊä³ö
print("array1ÅÅĞòÊä³ö")
for k,v in pairs(array1) do
	print(v)
end
--array2ÅÅĞòÊä³ö
print("array2ÅÅĞòÊä³ö")
for k,v in pairs(array2) do
	print(v)
end



